File: 1 - Hera's Quest for Power/solution.js
function findSecondLargestPower(n, powers) {
    let max = -1;
    let secondMax = -1;
    for (const power of powers) {
        if (power > max) {
            secondMax = max;
            max = power;
        } else if (power > secondMax && power < max) {
            secondMax = power;
        }
    }
    return secondMax;
}

// Read input and process it
process.stdin.resume();
process.stdin.setEncoding("utf-8");
let inputString = "";
process.stdin.on("data", (input) => {
    inputString += input;
});

process.stdin.on("end", () => {
    const inputLines = inputString.trim().split('\n');
    const t = parseInt(inputLines[0].trim(), 10);
    let line = 1;

    for (let i = 0; i < t; i++) {
        const n = parseInt(inputLines[line].trim(), 10);
        const powers = inputLines[line + 1].trim().split(' ').map(Number);
        line += 2;
        console.log(findSecondLargestPower(n, powers));
    }
});

File: 1 - Hera's Quest for Power/solution.py
def find_second_largest_power(N, powers):
    max_power = -1
    second_max_power = -1
    for i in range(N):
        if powers[i] > max_power:
            second_max_power = max_power
            max_power = powers[i]
        elif powers[i] > second_max_power and powers[i] < max_power:
            second_max_power = powers[i]
    return second_max_power

if __name__ == "__main__":
    T = int(input())
    for _ in range(T):
        N = int(input())
        powers = list(map(int, input().split()))
        print(find_second_largest_power(N, powers))
